front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/trainNameHelper.ts

```diff
@@ -9,4 +13,25 @@ const trainNameWithNum = (name: string, actualTrainCount: number, total: number)
   return `${name} ${actualTrainCount}`;
 };
 
+const normalizeString = (rollingStock: string): string =>
+  rollingStock.toUpperCase().replace(/[_\W]/g, '');
+
+export const findValidTrainNameKey = (
+  rollingStock: string
+): RollingstockOpenData2OSRDKeys | undefined => {
+  const normalizedRollingStock = normalizeString(rollingStock);
+
+  return Object.keys(rollingstockOpenData2OSRD).find((key) => {
+    const normalizedKey = normalizeString(key);
+    const normalizedValue = normalizeString(
+      rollingstockOpenData2OSRD[key as keyof typeof rollingstockOpenData2OSRD]
+    );
+
+    return (
+      normalizedRollingStock.includes(normalizedKey) ||
+      normalizedRollingStock.includes(normalizedValue)
```

Comment:
can you replace name as follow 
```
key => rollingstock
normalizedKey => normalizedRollingstock
noralizedValue => normalizedRollingStockValue
```
without the file stucture on the side it will be more comprehensible

Reply:
I let key as key, changed the others.